File: backend-example-python/src/application/test/test_ioc_class_3.py
class TestIocClass3:
    """IoC测试类3"""

    def __init__(self):
        """不会引起循环依赖的构造函数"""
        pass

    # from src.application.test.test_ioc_class_1 import TestIocClass1
    #
    # def __init__(self, class1: TestIocClass1):
    #     """会引起循环依赖的构造函数"""
    #     self.class1 = class1

File: backend-example-java/src/main/java/com/example/backend/application/test/TestIocClass3.java
package com.example.backend.application.test;

import org.springframework.stereotype.Component;

/**
 * IoC测试类3
 */
@Component
public class TestIocClass3 {

    /**
     * 不会引起循环依赖的构造函数
     */
    public TestIocClass3() {
    }

    //private final TestIocClass1 class1;
    //
    ///**
    // * 会引起循环依赖的构造函数
    // */
    //public TestIocClass3(TestIocClass1 class1) {
    //    this.class1 = class1;
    //}
}
